### PR TITLE
Maps Chapter 1

### PR DESCRIPTION
**Pull request by marshall-alex**
OpenLayers library now delivered via Cloudfire CDN instead of including JS Library in asset pipeline.

### CODE REVIEW

**Comment by goldenhub77** [comment]
@kgf @marshall-alex Hey Marshall, there are 24 tests that need to be resolved before this can be assessed and merged.

**Comment by kgf** [comment]
I wonder if this is in regard to the PR, or something else. The errors don't seem to be related to the areas of code changed...
Ideas ?

**Comment by kgf** [comment]
@goldenhub77 I have resolved the issue with the tests. Could you please take a look now ?

File: app/views/tasks/show.html.erb [hunk 1]
```diff
@@ -31,6 +32,46 @@
...
+    element: document.getElementById('marker'),
+    stopEvent: false
+  });
+  map.addOverlay(marker);
+
+  var task = new ol.Overlay({
+    position: pos,
+    element: document.getElementById('task')
+  });
+  map.addOverlay(task);
```
Comment (kgf) [inline app/views/tasks/show.html.erb:71]: What is the difference between the task overlay and the marker overlay ? Why are there two ?

Comment (marshall-alex) [reply app/views/tasks/show.html.erb:71]: the task overlay is the title of the event shown on the map and the marker overlay is the marker image

Comment (marshall-alex) [reply app/views/tasks/show.html.erb:71]: Let me see if I can condense them into one overlay

Comment (kgf) [reply app/views/tasks/show.html.erb:71]: Thanks ! Interesting, the task title isn't handled by line 38 ?

Comment (marshall-alex) [reply app/views/tasks/show.html.erb:71]: The task title is handled by line 38 as well as lines 67-71. The marker icon is handled on 39 and 60-65.

Comment (kgf) [reply app/views/tasks/show.html.erb:71]: Our comments got mingled. I don't see a value to combining them into one layer, and could see disadvantages, like wanting to turn off only the titles.

Comment (marshall-alex) [reply app/views/tasks/show.html.erb:71]: Okay got it. Is there anything more for this commit then?

Comment (kgf) [reply app/views/tasks/show.html.erb:71]: I'd like someone more competent in JS to take a look...

Comment (goldenhub77) [reply app/views/tasks/show.html.erb:71]: @marshall-alex @kgf Hey guys, I will take a look at this tomorrow, I was very busy today and was not able to code review.

Comment (kgf) [reply app/views/tasks/show.html.erb:71]: That's great, thanks for your help!

File: db/schema.rb [hunk 2]
```diff
@@ -173,7 +167,7 @@
     t.text     "comments"
     t.string   "error_code"
     t.string   "id_code"
-    t.boolean  "is_template", default: false
+    t.boolean  "is_template",                         default: false
```
Comment (goldenhub77) [inline db/schema.rb]: @marshall-alex unknown extra whitespace here.

File: db/schema.rb [hunk 3]
```diff
@@ -164,7 +158,7 @@
     t.string   "description"
     t.datetime "start_time"
     t.datetime "end_time"
-    t.integer  "duration"
+    t.decimal  "duration",    precision: 7, scale: 2
```
Comment (goldenhub77) [inline db/schema.rb]: @marshall-alex unknown extra whitespace here.

File: app/views/tasks/show.html.erb [hunk 4]
```diff
@@ -31,6 +32,46 @@
   <p>(<%= @task.latitude %>, <%= @task.longitude %>)</p>
 <% end %>
 
+<% if @task.latitude.present? and @task.longitude.present? %>
+  <div id="map" class="map" style="height:250px"></div>
+  <div style="display: none;">
+  <a class="overlay" id="task" target="_blank" href=""><%= @task.title %></a>
```
Comment (goldenhub77) [inline app/views/tasks/show.html.erb]: @marshall-alex class "overlay" does not appear to have a purpose as of now. Do you have one for this?

File: app/views/tasks/show.html.erb (hunk 1 above)
Comment (goldenhub77) [reply app/views/tasks/show.html.erb:71]: @marshall-alex @kgf After reviewing this, I don't see any major issues with the way this is implemented I just made a few comments with questions.

File: db/schema.rb (hunk 2 above)
Comment (kgf) [reply db/schema.rb]: Correct me if I'm wrong, @marshall-alex but I doubt you edited this file. It is typically system generated. I would like to know what causes this behavior. Did you perhaps open it in your editor, which did some auto formatting ?

Comment (marshall-alex) [reply db/schema.rb]: I did not purposely make any changes to that file I'd guess my editor is probably the culprit

File: app/views/tasks/show.html.erb (hunk 4 above)
Comment (marshall-alex) [reply app/views/tasks/show.html.erb]: No, sorry, I should've removed it. I ended up using the id instead and forgot to take out the class

File: db/schema.rb (hunk 3 above)
Comment (marshall-alex) [reply db/schema.rb]: This was also unintentional

**Comment by marshall-alex** [comment]
@goldenhub77 @kgf I'm going to remove the two instances of white space in the schema.rb file and take out the overlay class and push those changes in the next half hour. Thanks for all the help

**Comment by marshall-alex** [comment]
@goldenhub77 @kgf I've addressed the three comments here and the opacity issue from slack

**Pull request merged**